### ISSUE TITLE
Loading data using ajax in componentDidMount fails

### ISSUE THREAD

**Issue by iandanforth** [GriddleGriddle/Griddle#169]
I have a simple wrapper component around griddle that handles loading data from my API. Prior to the commit mentioned below it loads data successfully. After the commit, there is no error thrown, but data is not displayed.

```
    app.HomeComponent = React.createClass({

        getInitialState: function() {
            var initState = {
                dataSource: app.settings.baseAPI + "logs/",
                data: [{}],
            }

            return initState;
        },

        componentDidMount: function() {
            console.log(this.state.dataSource);
            $.get(this.state.dataSource, function(result) {
                if (this.isMounted()) {
                    this.setState({
                        data: result.rows
                    });
                }
            }.bind(this));
        },

        render: function() {

            return (
                <div className="grid-block">
                    <Griddle results={this.state.data} />
                </div>
            );
        }

    });
```

24ef50856a125c1f91acfd4150309b7de8b08d38 is the first bad commit
commit 24ef50856a125c1f91acfd4150309b7de8b08d38
Author: Ryan Lanciaux ryan.lanciaux@gmail.com
Date:   Mon Feb 2 00:53:55 2015 -0500

```
Makes more of the items in Griddle use the columnSettings. Breaks a lot of the tests.
```

:040000 040000 084aeb57617f1d472dddc90cfd3218fc05ba0de7 9063331873ce96c6e74e36ebc6442fe7833d96cb M  build
:040000 040000 076985d1dc4c5c95925469b9db2c9847388577a3 e0118e43153a315ed1758f13e6b635cae5e63d12 M  scripts

**Comment by bpugh** [comment]
This broke for me also. I haven't had a chance to look at exactly what changed but it seems to be an issue with changing results after the Griddle component is mounted. As a quick workaround this worked for me:

```
{this.state.data.length !== 0 ?
          <Griddle
            resultsPerPage={100}
            showPager={false}
            results={this.state.data}
            tableClassName="table"
          /> : 'No results to display'
 }

**Comment by aphillipo** [comment]
Ah - interesting, I'm too affected by this, I simply start off with stub state data that contains empty columns.

**Comment by aphillipo** [comment]
I've looked through that commit and it seems as though this.columnSettings is not passed as a prop so the render method in react does not know that things have changed.  Is this possible?  It looks like before that everything was set with props and so the normal React update methods worked.

Most of the time other things affect the results and it updated but it seems very very falkey and like some race condition is happening.

I really need to fix this bug so maybe going over to letting Griddle manage its own Ajax (using GriddleWithCallback) might fix things in the mean time.

Personally I think Griddle should try to avoid peoples data as much as is possible.

**Comment by aphillipo** [comment]
Looking at the code it seemed that it needs the "columns" prop set on the griddle component, then everything works.  Without it columns can under some strange circumstances end up blank/missing and so with 0 columns = no data displayed.  It's a bit weird what is going on here but I guess this at least fixes the issue for now.

Would be easy to do a patch that makes sure the "columns" prop is populated from the data but I imagine that code already exists somewhere and is not working for whatever reason.

Basically "columns" should be mandatory for a number of reasons including it fixing a different bug that I can't find now.

**Comment by aphillipo** [comment]
It was this bug about visibility ( columnMetadata: { visible: bool } does not work): https://github.com/GriddleGriddle/Griddle/issues/114

**Comment by ryanlanciaux** [comment]
We've changed the API in the latest version of Griddle and modified how column settings are being passed in to Griddle. This _should_ no longer be a problem in the new version but please feel free to open a new issue if any problems are encountered. 👍
